features/layout_data.feature

```diff
@@ -0,0 +1,18 @@
+Feature: Layout data
+  As a hacker who likes to avoid repetition
+  I want to be able to embed data into my layouts
+  In order to make the layouts slightly dynamic
+
+  Scenario: Use custom layout data
+    Given I have a _layouts directory
+    And I have a "_layouts/custom.html" file with content:
+      """
+      ---
+      foo: my custom data
+      ---
+      {{ content }} foo: {{ layout.foo }}
```

Comment:
In my case I was doing `{{ layout.foo }}` from parent layout file `index.html`, not from `custom.html`.